packages/engine/errors/8016.md

```diff
@@ -0,0 +1,6 @@
+---
+original: "Type assertion expressions can only be used in TypeScript files."
+excerpt: "You can't use type assertions because this isn't a Typescript file"
+---
+
+You're trying to use Typescript but I don't think the current file extension is appropriate.
```

Comment:
imho, I'm not sure this adds any value to understanding the original message:
```
...can only be used in TypeScript files.
```
This last part is very clear that the **thing** can only be used in TypeScript files. So the excerpt and description are saying the exact same thing with more words and less information, and aren't exactly describing any solution. 👀 

There are about ~12 diagnostic errors which end in the same way and the issue is not the extension but the TS language feature you're using in a JS file.

Raw ideas:
```suggestion
Type assertions such as `myVar as string` are a TypeScript-only language feature.

To continue using type assertions, convert your JavaScript file to TypeScript. Alternatively, remove the type assertions from your JavaScript file.
```

Reply:
Agree with your suggestion, didn't add the alternative solution because I'd assume if someone's using type assertions they intend on using typescript